dot/state/grandpa.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package state
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"math/big"
+
+	"github.com/ChainSafe/chaindb"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/scale"
+)
+
+var (
+	genesisSetID      = uint64(0)
+	grandpaPrefix     = "grandpa"
+	authoritiesPrefix = []byte("auth")
+	setIDChangePrefix = []byte("change")
+	pauseKey          = []byte("pause")
+	resumeKey         = []byte("resume")
+	currentSetIDKey   = []byte("setID")
+)
+
+// GrandpaState tracks information related to grandpa
+type GrandpaState struct {
+	baseDB chaindb.Database
+	db     chaindb.Database
```

Comment:
yes exactly, there's an issue open to clean this up: https://github.com/ChainSafe/gossamer/issues/1309 might just do this quick since it's a bit confusing atm